### DISCUSSION TITLE
Elevated rate of abandoned PRs

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#30741] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket Cloud

### Please tell us more about your question or problem

Running renovate version `38.25.0`

We're seeing an elevated amount of `abandoned` PRs.  

These PRs only have a single commit by our Renovate user, however the PR is left open with a comment

> Autoclosing Skipped
> 
> This PR has been flagged for autoclosing. However, it is being skipped due to the branch being already modified. Please close/delete it manually or report a bug if you think this is in error.

This mainly seems to be affecting our `archived` repositories, for which we have this shared preset package rule
```
{
  "description": "Skip any archived repositories",
  "matchRepositories": ["**/*-archived"],
  "enabled": false
},
```

Not sure if skipped repository PRs are being incorrectly classified as abandoned 🤔 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
I don't think the `packageRules` you set will have any effect, as the `packageRules` are not applied in the beginning. They are only useful once packages are extracted. You will need to add `enabled:false` either in each repo's config. Or, use `autodiscover` and `autodiscoverFilter` to exclude those repos. (this is to be placed in admin config ie. `config.js`)

**Reply by setchy** [reply]
> I don't think the `packageRules` you set will have any effect, as the `packageRules` are not applied in the beginning. They are only useful once packages are extracted. You will need to add `enabled:false` either in each repo's config. Or, use `autodiscover` and `autodiscoverFilter` to exclude those repos. (this is to be placed in admin config ie. `config.js`)

🤔 not sure what to say, the package rule does work for us 😉 

re: the abandoned PRs with a single commit from Renovate and no other changes, we have >~200 of them floating around.

**Reply by RahulGautamSingh** [reply]
1. If an existing PR from renovate is modified, it is intentional to leave a comment and let them be .. we do not close them. If they aren't modified and this still occurs for you...please share some logs.

 2. The repos seems disabled, but they are getting processed till extract phase. Then this `packageRule` gets applied and all the deps get disabled. The bot now has no deps to update so we skip to `finalizeRepo`. Which is where we try to autoclose open PRs. 🤔

**Reply by setchy** [reply]
An example

For this `-archived` repository we have 2 Renovate PRs marked as abandoned
![Screenshot 2024-08-13 at 1 06 22 PM](https://github.com/user-attachments/assets/4aca2175-2c3b-499a-b26d-d2de2205f790)
![Screenshot 2024-08-13 at 1 06 45 PM](https://github.com/user-attachments/assets/2c61f428-5715-47f8-b593-da61724ec2bb)
![Screenshot 2024-08-13 at 1 14 03 PM](https://github.com/user-attachments/assets/67b30c21-500d-45d9-b23a-b75a519c4cc8)

Logs excerpt
```
DEBUG: validateReconfigureBranch() (repository=org/repository-archived)
DEBUG: No reconfigure branch found (repository=org/repository-archived)
DEBUG: No renovate config file found (repository=org/repository-archived)
DEBUG: checkConfigMigrationBranch() (repository=org/repository-archived)
DEBUG: checkConfigMigrationBranch() Config does not need migration (repository=org/repository-archived)
DEBUG: Removing any stale branches (repository=org/repository-archived)
DEBUG: config.repoIsOnboarded=true (repository=org/repository-archived)
DEBUG: Branch lists (repository=org/repository-archived)
       "branchList": [],
       "renovateBranches": ["renovate/com.spotify-docker-maven-plugin-0.x", "renovate/cucumber.version"]
DEBUG: remainingBranches=renovate/com.spotify-docker-maven-plugin-0.x,renovate/cucumber.version (repository=org/repository-archived)
DEBUG: findPr(renovate/com.spotify-docker-maven-plugin-0.x, undefined, open) (repository=org/repository-archived)
DEBUG: getPrList() (repository=org/repository-archived)
DEBUG: Found PR #3 (repository=org/repository-archived)
DEBUG: branch.isModified(): using git to calculate (repository=org/repository-archived)
DEBUG: branch.isModified() = true (repository=org/repository-archived)
DEBUG: setCachedModifiedResult(): Branch cache not present (repository=org/repository-archived)
DEBUG: Branch is modified - skipping PR autoclosing (repository=org/repository-archived)
       "prNo": 3,
       "prTitle": "Update dependency com.spotify:docker-maven-plugin to v0.4.14 - abandoned"
DEBUG: Found 1 comments (repository=org/repository-archived)
DEBUG: Ensuring comment "Autoclosing Skipped" in #3 (repository=org/repository-archived)
DEBUG: Comment is already update-to-date (repository=org/repository-archived)
DEBUG: findPr(renovate/cucumber.version, undefined, open) (repository=org/repository-archived)
DEBUG: getPrList() (repository=org/repository-archived)
DEBUG: Found PR #2 (repository=org/repository-archived)
DEBUG: branch.isModified(): using git to calculate (repository=org/repository-archived)
DEBUG: branch.isModified() = true (repository=org/repository-archived)
DEBUG: setCachedModifiedResult(): Branch cache not present (repository=org/repository-archived)
DEBUG: Branch is modified - skipping PR autoclosing (repository=org/repository-archived)
       "prNo": 2,
       "prTitle": "Update cucumber.version to v1.2.6 - abandoned"
DEBUG: Found 1 comments (repository=org/repository-archived)
DEBUG: Ensuring comment "Autoclosing Skipped" in #2 (repository=org/repository-archived)
DEBUG: Comment is already update-to-date (repository=org/repository-archived)
DEBUG: Issues are disabled - cannot ensureIssueClosing (repository=org/repository-archived)
DEBUG: Issues are disabled - cannot ensureIssueClosing (repository=org/repository-archived)
DEBUG: PackageFiles.clear() - Package files deleted (repository=org/repository-archived)
DEBUG: getPrList() (repository=org/repository-archived)
DEBUG: Branch summary (repository=org/repository-archived)
       "cacheModified": undefined,
       "baseBranches": [{"branchName": "master", "sha": "bddb8c4389d65183d480833dca6ba913f8f1ade4"}],
       "branches": [],
       "defaultBranch": "master",
       "inactiveBranches": []
DEBUG: Renovate repository PR statistics (repository=org/repository-archived)
       "stats": {"total": 2, "open": 2, "closed": 0, "merged": 0}
DEBUG: Repository result: done, status: onboarded, enabled: true, onboarded: true (repository=org/repository-archived)
```

Looks like branch.isModified() is incorrectly detecting these as modified

**Reply by RahulGautamSingh** [reply]
Do you have trace logs?
If not, can you add this to your config and then share the logs after a run for this repo.
```json
{
 "logLevelRemap": {
 "matchMessage": "branch.isModified() = true",
 "newLogevel": "debug"
 }
}
```

**Reply by setchy** [reply]
To confirm, should it be remapped to `trace`?  
```
{
 "logLevelRemap": {
 "matchMessage": "branch.isModified() = true",
 "newLogevel": "trace"
 }
}

**Reply by RahulGautamSingh** [reply]
No. It's current log level is trace we want to change it to debug

**Reply by setchy** [reply]
Ah ha!
```
DEBUG: branch.isModified(): using git to calculate (repository=org/repository-archived)
TRACE: branch.isModified() = true (repository=org/repository-archived, baseBranch=master)
       "branchName": "renovate/com.spotify-docker-maven-plugin-0.x",
       "committedAuthors": ["renovate@whitesourcesoftware.com"],
       "includedAuthors": ["renovate@whitesourcesoftware.com"],
       "gitAuthorEmail": "renovatebot@our-org.com",
       "ignoredAuthors": []
DEBUG: branch.isModified() = true (repository=org/repository-archived)
```

Our global config
```
  "gitAuthor": "Renovate Bot <renovatebot@our-org.com>",
```

**Reply by setchy** [reply]
I might just add `renovate@whitesourcesoftware.com` to the `gitIgnoredAuthors` to unblock us

**Reply by setchy** [reply, marked as answer]
Confirmed that adding 
```
  "gitIgnoredAuthors": ["renovate@whitesourcesoftware.com"],
```

has resolved this.  Not sure how a bunch of commits were made with that author email.  🤷‍♂️ 

Thanks for your help @RahulGautamSingh 🙇

**Reply by setchy** [reply]
It does seem very unusual that we're now getting a mixture of commits with the default gitAuthor, and not our custom gitAuthor

**Reply by ssi-ips** [reply]
I noticed the same but the commiter is different:

```
2024-08-16T11:20:38.2760833Z TRACE: branch.isModified() = true (repository=Nexus-IPS/driverstore, baseBranch=main, branch=renovate/com.vaadin-24.x)
2024-08-16T11:20:38.2761020Z        "branchName": "renovate/com.vaadin-24.x",
2024-08-16T11:20:38.2761385Z        "committedAuthors": ["122013515+renovate-nexus-ips[bot]@users.noreply.github.com"],
2024-08-16T11:20:38.2761746Z        "includedAuthors": ["122013515+renovate-nexus-ips[bot]@users.noreply.github.com"],
2024-08-16T11:20:38.2761899Z        "gitAuthorEmail": "bot@renovateapp.com",
2024-08-16T11:20:38.2762043Z        "ignoredAuthors": ["bot@renovateapp.com"]
```

**Reply by dpozinen** [reply]
We're getting similar behaviour on `38.0.0` onwards. 

We first noticed this after not getting any PRs or auto merges due to the same `branch.isModified() = true`. After adding new ignored authors everything was fixed. But that doesn't explain why we are getting the commits from these other authors? All of our runners are setup with the same config of `gitAuthor` which is now ignored in the actual commit.

I see in the logs that it is set successfully:
```
DEBUG: Setting git author name: Renovate Bot (repository=..., branch=renovate/minor-dependencies)
DEBUG: Setting git author email: renovate@whitesourcesoftware.com (repository=..., branch=renovate/minor-dependencies)
```

but the resulting commit uses `<gh-app-id>+renovate[bot]@users.noreply.github.com` as author.
This then causes this logic:
https://github.com/renovatebot/renovate/blob/main/lib/util/git/index.ts#L679-L681

to not have any effect.

@setchy Have you been able to find the cause?:
> It does seem very unusual that we're now getting a mixture of commits with the default gitAuthor, and not our custom gitAuthor

**Reply by setchy** [reply]
> @setchy Have you been able to find the cause?:

I did not 🕵️

**Reply by dpozinen** [reply]
Could you verify that your logs also contain messages about the correct author being set in those wrong author commits?

**Reply by ymgyt** [reply]
We encountered the same issue.  
The root cause was that Renovate was being executed as a self-hosted GitHub App, so the commit's author was not the value specified as `gitAuthor`, but rather the one derived from the GitHub App's credentials(`<app-id>+<username>[bot]@users.noreply.github.github.com`).  

When `gitAuthor` is specified, it is used as the author's email for the `branch.isModified()` evaluation. However, that value was not actually applied to the commit itself.  

As a result, removing the `gitAuthor` from the configuration allowed `branch.isModified()` to evaluate correctly.
